index.rst

```diff
@@ -4,9 +4,164 @@ Writeable Butlers for Science Users
+If that turns out to be the case, we should probably drop temporary table support from the query system, as ``QuantumGraph`` generation was the driving user case for it.
+We should probably drop it from ``RemoteButler`` regardless, as emulated support is probably worse than no support.
+
+QuantumBackedButler Transfers from RemoteButler to DirectButler
+---------------------------------------------------------------
+
+After a ``QuantumGraph`` is generated against the official data repository, we need to be able to execute that graph such that:
+
+- input datasets are fetched from the ``RemoteButler`` to the offical data repository (with URL signing as needed);
+- output datasets are written into the datastore root of the personal data repository;
+- the "transfer" stage that inserts rows into the personal data repository's database can (optionally) transfer inputs as well as outputs, including setting up the collection structure in which the inputs were found in the offical data repository.
+
+This requires additional persistent information that would have to be put into the ``QuantumGraph`` or some other file.
+
+Signed URLs in DirectButler
+---------------------------
+
+In order for a personal data repository to reference an official data product (instead of copying it), the ``DirectButler`` client will need to be able to obtain signed URLs from the ``RemoteButler`` server.
+This could be implemented as a new ``Datastore`` subclass (to be chained in with a regular ``FileDatastore`` for the personal datasets in that repository), an extension to the current ``FileDatastore``.
+It could also be implemented directly within ``DirectButler``, but since we want this functionality in ``QuantumBackedButler`` as well, implementing it in ``Datastore`` seems to make more sense.
+
+User Interfaces
+---------------
+
+We need high-level command-line and notebook-friendly tools for at least the following tasks:
+
+- Create a new, empty satellite data repository from the official data repository.
+  This needs to set up a datastore that can reference official datasets (with URL signing), and it may be convenient for it to always register a few instruments and/or skymaps, and maybe even write curated calibrations.
+
+- Create and run a ``QuantumGraph`` with inputs from the official data repository or the personal data repository and outputs going to the personal data repository, using ``QuantumBackedButler``.
+  In the minimum-viable system the input source would strictly be one repository OR the other, but in the future we may be able to support combinations, and should plan for that in the UI.
+  I would recommend splitting this up into multiple subcommands, centered around a user-visible directory that holds the ``QuantumGraph`` file and some transfer metadata, similar to how BPS interacts with its "submit" directory.
```

Comment:
I think this is too obscure / too closely tied to something in the author's mind that hasn't been fully expressed here.  I *think* I know what you mean, but others will probably not if they haven't been embedded in this.

Reply:
Happy to drop it.  I agree it doesn't need to be in this document.